CRM/Core/Payment/AuthorizeNet.php

```diff
@@ -113,6 +113,10 @@ protected function supportsRecurContributionsForPledges() {
    * @throws \Civi\Payment\Exception\PaymentProcessorException
    */
   public function doPayment(&$params, $component = 'contribute') {
+    if ($params instanceof \Civi\Payment\PropertyBag) {
```

Comment:
I know this is your work in progress PR & we are not supposed to be looking at it but I stumbled here from your other PR & saw this - we can't actually have something that requires the payment processors to add something like this in what we actually merge

Reply:
I've put it there for now but we can also put it in the eventcart code just before calling `doPayment()` which would be a nice way of allowing us to pass a propertyBag to a paymentprocessor without requiring any modifications on the paymentprocessor side. Additionally a paymentprocessor could explicitly choose to switch ON legacy warnings once it's suitably converted.

Reply:
Yep - not requiring any modifications on the processor side is the pre-requisite for us to start passing out propertyBag from any core functions